src/backend/api/meals.js

```diff
@@ -2,13 +2,71 @@ const express = require("express");
 const router = express.Router();
 const knex = require("../database");
 
-router.get("/", async (request, response) => {
+// GET all meals
+router.get("/", async (req, res) => {
   try {
-    // knex syntax for selecting things. Look up the documentation for knex for further info
-    const titles = await knex("meals").select("title");
-    response.json(titles);
+    const meals = await knex("meals").select("*");
+    res.json(meals);
   } catch (error) {
-    throw error;
+    console.error(error.message);
+    res.status(500).send("Server Error");
+  }
+});
+
+// GET a single meal by id
+router.get("/:id", async (req, res) => {
+  try {
+    const meal = await knex("meals").where({ id: req.params.id }).first();
+    if (!meal) {
+      return res.status(404).json({ message: "Meal not found" });
+    }
+    res.json(meal);
+  } catch (error) {
+    console.error(error.message);
+    res.status(500).send("Server Error");
+  }
+});
+
+// CREATE a new meal
+router.post("/", async (req, res) => {
+  try {
+    const { name, description, price } = req.body;
+    const newMeal = await knex("meals").insert({ name, description, price });
+    res.status(201).json(newMeal);
+  } catch (error) {
+    console.error(error.message);
+    res.status(500).send("Server Error");
+  }
+});
+
+// UPDATE a meal by id
+router.put("/:id", async (req, res) => {
+  try {
+    const { name, description, price } = req.body;
```

Comment:
What if some of these haven't been provided, did you check if the old values are kept?